Week2/homework/map_filter.js

```diff
@@ -1,5 +1,69 @@
-'use strict';
-
 const numbers = [1, 2, 3, 4];
 
-// Add your code here
+for (let i = 0; i < numbers.length; i++) {
```

Comment:
This is not needed.  The `filter` and `map` methods look through the array automatically.
What you have done is loop through the array 4 * (1 (filter) + 1 (map)).  And that is why the results are shown 4 times.
This would do the trick:
```
const newNumbers = numbers.filter(number => number % 2);
const finalNumbers = newNumbers.map(number => number * 2);
console.log('The final numbers are ' + finalNumbers);
```
`map`, `filter` and `reduce` will be able to replace almost all your needs for `for`, `for...in`, and `while`, `do...while` etc. looping structures.